### PR TITLE
[DOC] Update docs for 23.12.1 release [skip ci]

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR updates download docs for 23.12.1 release.

Signed-off-by: Suraj Aralihalli <suraj.ara16@gmail.com>

### CODE REVIEW

**Review by viadea — APPROVED** [review]

File: docs/download.md [hunk 1]
```diff
@@ -97,6 +97,10 @@ New functionality and performance improvements for this release include:
 * Kernel optimizations have been implemented to improve Parquet read performance.
 * RAPIDS Accelerator also built and tested with Scala 2.13.
 * Last version to support Pascal-based Nvidia GPUs; discontinued in the next release.
+* Introduced support for parquet Legacy rebase mode (spark.sql.parquet.datetimeRebaseModeInRead=LEGACY and spark.sql.parquet.int96RebaseModeInRead=LEGACY)
+* Introduced support for Percentile function.
+* Delta lake 2.3 support.
+* Fixes a potential data corruption issue.
```
Comment (sameerz) [inline docs/download.md:103]: This comment is too vague.  Are we specifically referring to https://github.com/rapidsai/cudf/pull/14557 ?  I think that is a bit of a corner case and ok not to mention.  
```suggestion
```

**Review by viadea — DISMISSED** [review]

**Review by sameerz — APPROVED** [review]

**Comment by viadea** [comment]
build

**Review by viadea — APPROVED** [review]

**Comment by viadea** [comment]
build

**Comment by viadea** [comment]
build

**Pull request merged**